### PR TITLE
Extend contribution guidelines (new resource, datasource, service)

### PR DESCRIPTION
**Pull request by joaopalet**

### CODE REVIEW

File: CONTRIBUTION.md [hunk 1]
```diff
@@ -28,7 +29,353 @@ These commands can be executed from the project root:
...
+	var model Model
+	diags := req.Plan.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	projectId := model.ProjectId.ValueString()
+	barId := model.BarId.ValueString()
+	ctx = tflog.SetField(ctx, "project_id", projectId)
+	ctx = tflog.SetField(ctx, "bar_id", barId)
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:208]: usually this is only set after getting the response, where the ID is returned

File: CONTRIBUTION.md [hunk 2]
```diff
@@ -28,7 +29,353 @@ These commands can be executed from the project root:
...
+				},
+				Validators: []validator.String{
+					// Validators can be used to validate the values set to a field
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"bar_id": schema.StringAttribute{
+				Description: descriptions["bar_id"],
+				Required:    true,
```
[the remark this answers is not in the document]
Comment (joaopalet) [reply CONTRIBUTION.md:172]: Good catch, I had it as "name" and then changed to "ID" and forgot to update

File: CONTRIBUTION.md (hunk 1 above)
Comment (joaopalet) [reply CONTRIBUTION.md:208]: Good catch, same as https://github.com/stackitcloud/terraform-provider-stackit/pull/322#discussion_r1557657006

File: CONTRIBUTION.md [hunk 3]
```diff
@@ -28,7 +29,353 @@ These commands can be executed from the project root:
...
+				PlanModifiers: []planmodifier.String{
+					// "RequiresReplace" makes the provider recreate the resource when the field is changed in the configuration
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"my_field": schema.StringAttribute{
```
[the remark this answers is not in the document]
Comment (joaopalet) [reply CONTRIBUTION.md]: Added a required field

File: CONTRIBUTION.md [hunk 4]
```diff
@@ -33,22 +32,22 @@ These commands can be executed from the project root:
...
-   2. Please refer to the [Resource file structure](./CONTRIBUTION.md/#resource-file-structure) section for details on the strcutre of the file itself
+   1. The Go package should be similar to the service name, in this case `foo` would be an adequate package name
+   2. Please refer to the [Resource file structure](./CONTRIBUTION.md/#resource-file-structure) section for details on the structure of the file itself
 3. To register the new resource `bar` in the provider, add it to the `Resources` in `stackit/provider.go`, using the `NewBarResource` method
 4. Add an example in `examples/resources/stackit_foo_bar/resource.tf` with an example configuration for the new resource, e.g.:
    ```tf
     resource "stackit_foo_bar" "example" {
       project_id    = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
-      my_field      = "my-field-value"
+      my_optional_field      = "my-field-value"
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:50]: here it should also have the required field

File: CONTRIBUTION.md [hunk 5]
```diff
@@ -169,7 +172,7 @@ func (r *barResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *
 			},
 			"bar_id": schema.StringAttribute{
 				Description: descriptions["bar_id"],
-				Required:    true,
+				Computed:    true,
 				PlanModifiers: []planmodifier.String{
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:176]: remove the planModifier, it's not needed anymore

File: CONTRIBUTION.md [hunk 6]
```diff
@@ -179,7 +182,10 @@ func (r *barResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *
 					validate.NoSeparator(),
 				},
 			},
-			"my_field": schema.StringAttribute{
+			"my_required_field": schema.StringAttribute{
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:185]: here you could add the RequiresReplace

File: CONTRIBUTION.md [hunk 7]
```diff
@@ -244,7 +250,7 @@ func (r *barResource) Read(ctx context.Context, req resource.ReadRequest, resp *
 	projectId := model.ProjectId.ValueString()
 	barId := model.BarId.ValueString()
 	ctx = tflog.SetField(ctx, "project_id", projectId)
-	ctx = tflog.SetField(ctx, "my_field", barId)
+	ctx = tflog.SetField(ctx, "barId", barId)
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:254]: it should be "bar_id"

**Review by vicentepinto98 — APPROVED** [review]

**Pull request merged**